framework/Source/RawDataOutput.swift

```diff
@@ -28,13 +31,25 @@ public class RawDataOutput: ImageConsumer {
 
         renderFramebuffer.activateFramebufferForRendering()
         clearFramebufferWithColor(Color.Black)
-        renderQuadWithShader(sharedImageProcessingContext.passthroughShader, uniformSettings:ShaderUniformSettings(), vertices:standardImageVertices, inputTextures:[framebuffer.texturePropertiesForOutputRotation(.NoRotation)])
+        if pixelFormat == .Luminance {
+            privatePixelFormat = PixelFormat.RGBA
+            let luminanceShader = crashOnShaderCompileFailure("RawDataOutput"){try sharedImageProcessingContext.programForVertexShader(defaultVertexShaderForInputs(1), fragmentShader:LuminanceFragmentShader)}
+            renderQuadWithShader(luminanceShader, vertices:standardImageVertices, inputTextures:[framebuffer.texturePropertiesForTargetOrientation(renderFramebuffer.orientation)])
+        } else {
+            privatePixelFormat = pixelFormat
+            renderQuadWithShader(sharedImageProcessingContext.passthroughShader, uniformSettings:ShaderUniformSettings(), vertices:standardImageVertices, inputTextures:[framebuffer.texturePropertiesForOutputRotation(.NoRotation)])
+        }
         framebuffer.unlock()
         
         var data = [UInt8](count:Int(framebuffer.size.width * framebuffer.size.height * 4), repeatedValue:0)
         glReadPixels(0, 0, framebuffer.size.width, framebuffer.size.height, GLenum(GL_RGBA), GLenum(GL_UNSIGNED_BYTE), &data)
         renderFramebuffer.unlock()
 
-        dataAvailableCallback?(data)
+        if dataAvailableCallback != nil {
```

Comment:
You already using [https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/OptionalChaining.html](optional chaining) when calling dataAvailableCallback closure, so there is no need to check against nil value ;)